src/cli/service/echo.rs

```diff
@@ -269,43 +268,15 @@ where
         mut self,
         executor: Executor,
     ) -> Result<impl Service<(), TcpStream, Response = (), Error = Infallible>, BoxError> {
-        let (tcp_forwarded_layer, http_forwarded_layer) = match &self.forward {
-            None => (None, None),
-            Some(ForwardKind::Forwarded) => (
-                None,
-                Some(Either7::A(GetForwardedHeadersLayer::forwarded())),
-            ),
-            Some(ForwardKind::XForwardedFor) => (
-                None,
-                Some(Either7::B(GetForwardedHeadersLayer::x_forwarded_for())),
-            ),
-            Some(ForwardKind::XClientIp) => (
-                None,
-                Some(Either7::C(GetForwardedHeadersLayer::<XClientIp>::new())),
-            ),
-            Some(ForwardKind::ClientIp) => (
-                None,
-                Some(Either7::D(GetForwardedHeadersLayer::<ClientIp>::new())),
-            ),
-            Some(ForwardKind::XRealIp) => (
-                None,
-                Some(Either7::E(GetForwardedHeadersLayer::<XRealIp>::new())),
-            ),
-            Some(ForwardKind::CFConnectingIp) => (
-                None,
-                Some(Either7::F(GetForwardedHeadersLayer::<CFConnectingIp>::new())),
-            ),
-            Some(ForwardKind::TrueClientIp) => (
-                None,
-                Some(Either7::G(GetForwardedHeadersLayer::<TrueClientIp>::new())),
-            ),
-            Some(ForwardKind::HaProxy) => (Some(HaProxyLayer::default()), None),
+        let tcp_forwarded_layer = match &self.forward {
+            Some(ForwardKind::HaProxy) => Some(HaProxyLayer::default()),
+            _ => None,
         };
 
```

Comment:
```suggestion

let http_service = self.build_http();

```

move it to here, so that you do not need to clone `cfg` :)